lib/src/repo_path.rs

```diff
@@ -83,6 +83,24 @@ impl RepoPath {
         }
     }
 
+    /// Converts workspace-relative `Path` to `RepoPath`.
+    ///
+    /// The input path should not contain `.` or `..`.
+    pub fn from_relative_path(relative_path: impl AsRef<Path>) -> Result<Self, FsPathParseError> {
+        let relative_path = relative_path.as_ref();
+        let components = relative_path
+            .components()
+            .map(|c| match c {
+                Component::Normal(a) => {
```

Comment:
[comment, non-blocking] Does `a` stand for anything? I personally like to shadow the variables in `match`es like this and would have written `c` again, but I don't feel strongly.

Reply:
Ok, I'll rename it to `name` in the other PR. I'm going to rewrite `RepoPath` to store components in `String` instead of `Vec<String>`, and this PR will conflict with that.

I tend to avoid shadowing with unrelated type, but it's just personal preference.